Scripts/RemediationScripts/Remediate-PTAControls.ps1

```diff
@@ -0,0 +1,915 @@
+       
+###>
+
+
+function Setup-Prerequisites
+{
+    <#
+        .SYNOPSIS
+        Checks if the prerequisites are met, else, sets them up.
+
+        .DESCRIPTION
+        Checks if the prerequisites are met, else, sets them up.
+        Includes installing any required Azure modules.
+
+        .INPUTS
+        None. You cannot pipe objects to Setup-Prerequisites.
+
+        .OUTPUTS
+        None. Setup-Prerequisites does not return anything that can be piped and used as an input to another command.
+
+        .EXAMPLE
+        PS> Setup-Prerequisites
+
+        .LINK
+        None
+    #>
+
+    # List of required modules
+    $requiredModules = @("Az.Accounts", "Az.Resources", "AzureAD")
+
+    Write-Host "Required modules: $($requiredModules -join ', ')" -ForegroundColor $([Constants]::MessageType.Info)
+    Write-Host "Checking if the required modules are present..."
+    # List of available modules
+    $availableModules = $(Get-Module -ListAvailable $requiredModules -ErrorAction Stop)
+
+    # Check if the required modules are installed.
+    if ($availableModules.Name -contains "Az.Resources")
+    {
+        $module = get-module "Az.Resources"
+        if($module.Version -ge "5.0.0")
+        {
+            Write-Host " Az.Resources module is present ..." -ForegroundColor $([Constants]::MessageType.Info)        
+        }
+        else
+        {
+            Write-Host "Installing $($_) module with required version(5.0.0)..." -ForegroundColor $([Constants]::MessageType.Info)
+            Install-Module -Name "Az.Resources" -MinimumVersion 5.0.0 -RequiredVersion 5.0.0 -Scope CurrentUser -Repository 'PSGallery'  -Force -ErrorAction Stop   
+        }
+    }
+    else
+    {
+        Write-Host "Installing $($_) module...." -ForegroundColor $([Constants]::MessageType.Update)
+        Install-Module -Name "Az.Resources" -MinimumVersion 5.0.0 -RequiredVersion 5.0.0 -Scope CurrentUser -Repository 'PSGallery' -Force -ErrorAction Stop
+    }
+
+
+    if($availableModules.Name -contains "AZ.Accounts")
+    {
+        $module = get-module "Az.Accounts"
+        if($module.Version -ge "2.5.4")
+        {
+            Write-Host "Az.Accounts module is present ... " -ForegroundColor $([Constants]::MessageType.Info)
+        }
+        else
+        {
+             Write-Host "Installing Az.Accounts module with required version(2.5.4)..." -ForegroundColor $([Constants]::MessageType.Info)
+             Install-Module -Name "Az.Accounts" -MinimumVersion 2.5.4 -RequiredVersion 2.5.4 -Scope CurrentUser -Repository 'PSGallery' -Force -ErrorAction Stop        
+        }    
+    }
+    else
+    {
+        Write-Host "Installing Az.Accounts module...." -ForegroundColor $([Constants]::MessageType.Update)
+        Install-Module -Name "Az.Accounts" -MinimumVersion 2.5.4 -RequiredVersion 2.5.4 -Scope CurrentUser -Repository 'PSGallery'  -Force -ErrorAction Stop
+    }
+
+    if($availableModules.Name -contains "AzureAD")
```

Comment:
Instead of specially handling this, can we modify L80 to include minimum required versions for each of the modules and use them here.